src/main/java/com/service/PersonService.java

```diff
@@ -93,4 +97,24 @@ public void showPersons() {
             System.out.println(person.toString());
         }
     }
+
+    @Override
+    public void showPersonsFromFile() {
+
+        String fileLoc = "C:\\Петко\\udemy\\java master class\\Martin_Project\\personsList.txt";
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileLoc))) {
+
+            String personLine;
+
+            while ((personLine = bufferedReader.readLine()) != null) {
+                String[] data = personLine.split(",");
```

Comment:
readData is more appropriate naming

Reply:
Also what happens if the line is empty and you try to split it ? I think this might answer your question